[ray-launcher] enable testing ray launcher on python 3.9

jieru-hu opened the pull request:
Closes #1205

Jasha10 commented on plugins/hydra_ray_launcher/integration_test_tools/create_integration_test_ami.py:
```diff
@@ -29,7 +29,7 @@ def _run_command(command: str) -> str:
 
 
-@hydra.main(config_name="create_integration_test_ami_config")
+@hydra.main(config_path=None, config_name="create_integration_test_ami_config")
```
This change is unrelated to python3.9 support, right?

jieru-hu replied:
Yes, this is just to fix the warning.

pixelb reviewed (approved):
CI is failing in ssh. I presume that's just a temporary issue

jieru-hu commented:
thanks for the reviews!

Added some docs for the testing tools, and also added even more logging to the tests so it's a bit easier to debug. All super minor. merging now...
